core/deployment/src/main/java/io/quarkus/artemis/core/deployment/ArtemisCoreProcessor.java

```diff
@@ -42,16 +45,44 @@ public class ArtemisCoreProcessor {
     private static final Logger LOGGER = Logger.getLogger(ArtemisCoreProcessor.class);
     private static final String FEATURE = "artemis-core";
 
-    static final Class<?>[] BUILTIN_CONNECTOR_FACTORIES = {
-            NettyConnectorFactory.class
-    };
-
-    static final Class<?>[] BUILTIN_LOADBALANCING_POLICIES = {
-            FirstElementConnectionLoadBalancingPolicy.class,
-            RandomConnectionLoadBalancingPolicy.class,
-            RandomStickyConnectionLoadBalancingPolicy.class,
-            RoundRobinConnectionLoadBalancingPolicy.class
-    };
+    static final List<ReflectiveClassBuildItem> BUILTIN_CONNECTOR_FACTORIES_REFLECTION_CONFIG = List.of(
+            ReflectiveClassBuildItem
+                    .builder(NettyConnectorFactory.class)
+                    .fields(false)
+                    .methods(false)
+                    .build(),
+            ReflectiveClassBuildItem
+                    .builder(ActiveMQConnectionFactory.class)
+                    .fields(false)
+                    .methods(true)
+                    .build(),
+            ReflectiveClassBuildItem
+                    .builder(HornetQClientProtocolManagerFactory.class)
+                    .fields(false)
+                    .methods(false)
+                    .build());
+
+    static final List<ReflectiveClassBuildItem> BUILTIN_LOADBALANCING_POLICIES_REFLECTION_CONFIG = List.of(
+            ReflectiveClassBuildItem
+                    .builder(FirstElementConnectionLoadBalancingPolicy.class)
+                    .fields(false)
+                    .methods(false)
+                    .build(),
+            ReflectiveClassBuildItem
+                    .builder(RandomConnectionLoadBalancingPolicy.class)
+                    .fields(false)
+                    .methods(false)
+                    .build(),
+            ReflectiveClassBuildItem
+                    .builder(RandomStickyConnectionLoadBalancingPolicy.class)
+                    .fields(false)
+                    .methods(false)
+                    .build(),
+            ReflectiveClassBuildItem
+                    .builder(RoundRobinConnectionLoadBalancingPolicy.class)
+                    .fields(false)
+                    .methods(false)
+                    .build());
```

Comment:
Is it possible to find all the classes which implement `ConnectionLoadBalancingPolicy` ?

Reply:
Depends on how you define "possible". Without JPMs or a Jandex index over the whole (runtime) application, however, there is (as far as I know) no efficient way to do this.

Reply:
Hmm, I forget the Jandex.

Reply:
Btw: if there is a(n efficient) way to do this, please let me know. We have a similar use case in [citrus (`github.com`)](https://github.com/citrusframework/citrus) (we need to find all classes annotated with a certain annotation).